pkg/kubelet/kubelet.go

```diff
@@ -1089,6 +1089,11 @@ func (kl *Kubelet) initialNodeStatus() (*api.Node, error) {
 		}
 	} else {
 		node.Spec.ExternalID = kl.hostname
+		// If no cloud provider is defined - use the one detected by cadvisor
+		info, err := kl.GetCachedMachineInfo()
```

Comment:
cc @roberthbailey: cAdvisor is already using the local metadata server to access GCE specific information. Relevant logic [here](https://github.com/google/cadvisor/blob/eab201fdf51c033299c8f4b43c13aec5d565d14c/utils/cloudinfo/gce.go).